src/saltext/vmware/utils/vm.py

```diff
@@ -789,3 +789,125 @@ def change_boot_options(vm, input_opts):
     utils_common.wait_for_task(task, vm.name, "configure boot options")
 
     return {"status": "changed"}
+
+
+def create_snapshot(vm_ref, snapshot_name, description, memory, quiesce):
+    """
+    Create virtual machine snapshot.
+
+    vm_ref
+        Reference to virtual machine.
+
+    snapshot_name
+        The name for the snapshot being created. Not unique
+
+    description
+        Description for the snapshot.
+
+    memory
+        (boolean) If TRUE, a dump of the internal state of the virtual machine (basically a memory dump) is included in the snapshot.
+
+    quiesce
+        (boolean) If TRUE and the virtual machine is powered on when the snapshot is taken, VMware Tools is used to quiesce the file system in the virtual machine.
+    """
+    task = vm_ref.CreateSnapshot_Task(snapshot_name, description, memory, quiesce)
+    ret = utils_common.wait_for_task(task, vm_ref.name, "create snapshot")
+    return ret
+
+
+def snapshot_recursive(snapshot_tree_root, snaps):
+    """
+    Recursively appends all snapshots in a snapshot tree.
+
+    snapshot_tree_root
+        Root node of snapshot tree.
+
+    snaps
+        List of snapshot info.
+    """
+    for ss in snapshot_tree_root:
+        current = {
+            "creation_time": str(ss.createTime),
+            "description": ss.description,
+            "id": ss.id,
+            "name": ss.name,
+            "state": ss.state,
+            "quiesced": ss.quiesced,
+        }
+        snaps.append(current)
+        if len(ss.childSnapshotList) > 0:
+            snaps = snapshot_recursive(ss.childSnapshotList, snaps)
+    return snaps
+
+
+def snapshot_recursive_search(snapshot_tree_root, snapshot_name, snapshot_id):
+    """
+    Recursively appends all snapshots in a snapshot tree.
+
+    snapshot_tree_root
+        Root node of snapshot tree.
+
+    snapshot_name
+        Name of snapshot to find.
+
+    snapshot_id
+        id of snapshot to find.
+    """
+    for ss in snapshot_tree_root:
+        if snapshot_id is None:
+            if snapshot_name == ss.name:
+                return ss
+            elif len(ss.childSnapshotList) > 0:
```

Comment:
**suggestion** similar suggestion here, and throughout. Rather than `len(...) > 0` just use the boolean nature of empty collections.